osu.Game/Screens/Play/HUD/DefaultRankDisplay.cs

```diff
@@ -37,9 +44,66 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            rank.Rank = scoreProcessor.Rank.Value;
+            switch (RankDisplay.Value)
+            {
+                case RankDisplayMode.Standard:
+                    rank.Rank = scoreProcessor.Rank.Value;
+                    break;
+
+                case RankDisplayMode.MinimumAchievable:
+                    rank.Rank = scoreProcessor.MinimumRank.Value;
+                    break;
+
+                case RankDisplayMode.MaximumAchievable:
+                    rank.Rank = scoreProcessor.MaximumRank.Value;
+                    break;
+            }
+            RankDisplay.BindValueChanged(mode =>
+            {
+                switch (mode.OldValue)
+                {
+                    case RankDisplayMode.Standard:
+                        scoreProcessor.Rank.UnbindBindings();
+                        break;
+
+                    case RankDisplayMode.MinimumAchievable:
+                        scoreProcessor.MinimumRank.UnbindBindings();
+                        break;
+
+                    case RankDisplayMode.MaximumAchievable:
+                        scoreProcessor.MaximumRank.UnbindBindings();
+                        break;
```

Comment:
This cannot be done. Calling `.UnbindBindings()` unilaterally on a public bindable like that will screw up other consumers.

Instead, create a new bindable local to this component, and bind it to / unbind it from `ScoreProcessor` bindables as required. (That is, if those continue to exist even.)